builtin/providers/ibmcloud/resource_ibmcloud_infra_virtual_guest.go

```diff
@@ -0,0 +1,1061 @@
+			GetPublicSubnets()
+		if err != nil {
+			log.Printf("Error getting secondary Ip addresses: %s", err)
+		}
+
+		secondaryIps := make([]string, 0)
+		for _, subnet := range secondarySubnetResult {
+			// Count static secondary ip addresses.
+			if *subnet.SubnetType == "STATIC_IP_ROUTED" {
+				for _, ipAddressObj := range subnet.IpAddresses {
+					secondaryIps = append(secondaryIps, *ipAddressObj.IpAddress)
+				}
+			}
+		}
+		if len(secondaryIps) > 0 {
+			d.Set("secondary_ip_addresses", secondaryIps)
+			d.Set("secondary_ip_count", len(secondaryIps))
+		}
+	}
+
+	return nil
+}
+
+func resourceIBMCloudInfraVirtualGuestUpdate(d *schema.ResourceData, meta interface{}) error {
+	sess := meta.(ClientSession).SoftLayerSession()
+	service := services.GetVirtualGuestService(sess)
+
+	id, err := strconv.Atoi(d.Id())
+	if err != nil {
+		return fmt.Errorf("Not a valid ID, must be an integer: %s", err)
+	}
+
+	result, err := service.Id(id).GetObject()
+	if err != nil {
+		return fmt.Errorf("Error retrieving virtual guest: %s", err)
+	}
+
+	// Update "hostname" and "domain" fields if present and changed
+	// Those are the only fields, which could be updated
+	if d.HasChange("hostname") || d.HasChange("domain") {
+		result.Hostname = sl.String(d.Get("hostname").(string))
+		result.Domain = sl.String(d.Get("domain").(string))
+
+		_, err = service.Id(id).EditObject(&result)
+
+		if err != nil {
+			return fmt.Errorf("Couldn't update virtual guest: %s", err)
+		}
+	}
+
+	// Set user data if provided and not empty
+	if d.HasChange("user_metadata") {
+		_, err := service.Id(id).SetUserMetadata([]string{d.Get("user_metadata").(string)})
+		if err != nil {
+			return fmt.Errorf("Couldn't update user data for virtual guest: %s", err)
+		}
+	}
+
+	// Update tags
+	if d.HasChange("tags") {
+		err := setGuestTags(id, d, meta)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Upgrade "cores", "memory" and "network_speed" if provided and changed
+	upgradeOptions := map[string]float64{}
+	if d.HasChange("cores") {
+		upgradeOptions[product.CPUCategoryCode] = float64(d.Get("cores").(int))
+	}
+
+	if d.HasChange("memory") {
+		memoryInMB := float64(d.Get("memory").(int))
+
+		// Convert memory to GB, as softlayer only allows to upgrade RAM in Gigs
+		// Must be already validated at this step
+		upgradeOptions[product.MemoryCategoryCode] = float64(int(memoryInMB / 1024))
+	}
+
+	if d.HasChange("network_speed") {
+		upgradeOptions[product.NICSpeedCategoryCode] = float64(d.Get("network_speed").(int))
+	}
+
+	if len(upgradeOptions) > 0 {
+		_, err = virtual.UpgradeVirtualGuest(sess, &result, upgradeOptions)
+		if err != nil {
+			return fmt.Errorf("Couldn't upgrade virtual guest: %s", err)
+		}
+
+		// Wait for softlayer to start upgrading...
+		_, err = WaitForUpgradeTransactionsToAppear(d, meta)
+
+		// Wait for upgrade transactions to finish
+		_, err = WaitForNoActiveTransactions(d, meta)
```

Comment:
It is is intentional in the sense that when we destroy the virtual guest for example, it will not allow saying that there are some Active transactions.